src/util/SimulatedAnnealing.hh

```diff
@@ -0,0 +1,185 @@
+#ifndef __SIMULATEDANNEALING__
```

Comment:
Could you add some brief documentation to the top of this file, and also use the conventional `__RAT_SimulatedAnnealing__` type header guards?